book/src/developers/designs/application_passwords.md

```diff
@@ -0,0 +1,188 @@
+# Rationale
+
+Kanidm exists to provide an authentication source for external applications.
+These applications need to have standardised ways to integrate with Kanidm to
+allow that application to interact and trust Kanidm's authentication results.
+
+For web based applications we offer Oauth2/OIDC. For Linux machines we offer a
+Kanidm specific HTTPS channel for identifying users. We also offer an LDAPS
+gateway for applications that don't support other protocols but we want users to
+use their "UNIX password".
```

Comment:
The point of this is also to remove the ability for unix password to do anything outside of unix integration, that way we can remove this from ldap. :)